src/boolean3.cpp

```diff
@@ -451,41 +341,55 @@ struct Kernel12 {
   }
 };
 
-std::tuple<Vec<int>, Vec<vec3>> Intersect12(
-    const Manifold::Impl &inP, const Manifold::Impl &inQ, const Vec<int> &s02,
-    const SparseIndices &p0q2, const Vec<int> &s11, const SparseIndices &p1q1,
-    const Vec<double> &z02, const Vec<vec4> &xyzz11, SparseIndices &p1q2,
-    bool forward) {
+std::tuple<Vec<int>, Vec<vec3>> Intersect12(const Manifold::Impl &inP,
+                                            const Manifold::Impl &inQ,
+                                            SparseIndices &p1q2, double expandP,
+                                            bool forward) {
   ZoneScoped;
   Vec<int> x12(p1q2.size());
   Vec<vec3> v12(p1q2.size());
 
-  for_each_n(
-      autoPolicy(p1q2.size(), 1e4), countAt(0_uz), p1q2.size(),
-      Kernel12({x12, v12, p0q2.AsVec64(), s02, z02, p1q1.AsVec64(), s11, xyzz11,
-                inP.halfedge_, inQ.halfedge_, inP.vertPos_, forward, p1q2}));
+  F02 f02{inP.vertPos_,
+          inQ.halfedge_,
+          inQ.vertPos_,
+          expandP,
+          forward ? inP.vertNormal_ : inQ.vertNormal_,
+          forward};
+  F11 f11{forward ? inP.vertPos_ : inQ.vertPos_,
```

Comment:
I will definitely fix these later, it is just interesting that it works within 30 minutes of trying :)